flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecSink.java

```diff
@@ -280,4 +326,64 @@ protected Transformation<RowData> applyUpsertMaterialize(
         materializeTransform.setStateKeyType(keySelector.getProducedType());
         return materializeTransform;
     }
+
+    private OneInputStreamOperator<RowData, RowData> createSumOperator(
+            ExecNodeConfig config,
+            RowType physicalRowType,
+            int[] inputUpsertKey,
+            GeneratedRecordEqualiser upsertKeyEqualiser,
+            GeneratedHashFunction upsertKeyHashFunction,
+            StateTtlConfig ttlConfig,
+            GeneratedRecordEqualiser rowEqualiser,
+            GeneratedHashFunction rowHashFunction) {
+
+        SinkUpsertMaterializeStrategy sinkUpsertMaterializeStrategy =
+                Optional.ofNullable(upsertMaterializeStrategy)
+                        .orElse(SinkUpsertMaterializeStrategy.LEGACY);
+
+        return sinkUpsertMaterializeStrategy == SinkUpsertMaterializeStrategy.LEGACY
+                ? SinkUpsertMaterializer.create(
+                        ttlConfig,
+                        physicalRowType,
+                        rowEqualiser,
+                        upsertKeyEqualiser,
+                        inputUpsertKey)
+                : SinkUpsertMaterializerV2.create(
+                        physicalRowType,
+                        rowEqualiser,
+                        upsertKeyEqualiser,
+                        rowHashFunction,
+                        upsertKeyHashFunction,
+                        inputUpsertKey,
+                        createStateConfig(
+                                sinkUpsertMaterializeStrategy,
+                                TimeDomain.EVENT_TIME,
+                                ttlConfig,
+                                config));
+    }
+
+    private static SequencedMultiSetStateConfig createStateConfig(
+            SinkUpsertMaterializeStrategy strategy,
+            TimeDomain ttlTimeDomain,
+            StateTtlConfig ttlConfig,
+            ReadableConfig config) {
+        switch (strategy) {
+            case VALUE:
+                return SequencedMultiSetStateConfig.forValue(ttlTimeDomain, ttlConfig);
+            case MAP:
+                return SequencedMultiSetStateConfig.forMap(ttlTimeDomain, ttlConfig);
+            case ADAPTIVE:
+                return SequencedMultiSetStateConfig.adaptive(
+                        ttlTimeDomain,
+                        config.getOptional(
+                                        TABLE_EXEC_SINK_UPSERT_MATERIALIZE_ADAPTIVE_THRESHOLD_HIGH)
+                                .orElse(null),
+                        config.getOptional(
+                                        TABLE_EXEC_SINK_UPSERT_MATERIALIZE_ADAPTIVE_THRESHOLD_LOW)
+                                .orElse(null),
```

Comment:
```suggestion
                        config.get(TABLE_EXEC_SINK_UPSERT_MATERIALIZE_ADAPTIVE_THRESHOLD_HIGH),
                        config.get(TABLE_EXEC_SINK_UPSERT_MATERIALIZE_ADAPTIVE_THRESHOLD_LOW),
```

It they are not set, and no default value, get will return null directly.